src/FirebaseArduino.h

```diff
@@ -37,7 +37,7 @@ class FirebaseArduino {
    * \param host Your firebase db host, usually X.firebaseio.com.
    * \param auth Optional credentials for the db, a secret or token.
    */
-  void begin(const String& host, const String& auth = "");
+  virtual void begin(const String& host, const String& auth = "");
```

Comment:
curious why this need to be virtual now?

Reply:
- Before virtual functions were in Firebase.h/FirebaseCall/FirebaseStream -> so they can be mocked in mock-firebase.h. Since that went away and modem library had to use FirebaseArduino instead, interface that needed to be mocked had to be declared virtual. 
So if modem library rests in peace -> (or just the unit tests for it) -> it won't be needed.